src/LibraryManager.Vsix/Commands/InstallLibraryCommand.cs

```diff
@@ -143,17 +143,11 @@ private async Task ExecuteAsync(object sender, EventArgs e)
                     }
                 }
 
-                UI.InstallDialog dialog = new UI.InstallDialog(dependencies, _libraryCommandService, configFilePath, target, rootFolder, project);
-
-                var dte = (DTE)Package.GetGlobalService(typeof(SDTE));
-                int hwnd = dte.MainWindow.HWnd;
-                WindowInteropHelper windowInteropHelper = new WindowInteropHelper(dialog);
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Since the if (project == null) case and this one both switch to the UI thread, move one of them above the if and get rid of the other one.

Reply:
Looks like this SwitchToMainThreadAsync isn't even needed since we're in the (project != null) block which has SwitchToMainThreadAsync already